// common-notification-service/common-notification-service.js
// Общая библиотека Notification
/* SQL Queries */
const SQL = {
  NOTIFICATION: {
    GET_PRODUCT_MAIL : `
    SELECT prw.* FROM pa_notification_service.product_mails prw WHERE 1=1  
	     and prw.product_id=$1
	     and to_user_id<>0
	     and prw.blocked is null and prw.deleted is null 
	  order by created asc`,
    GET_PRODUCT_MAIL_PERSONAL: `
    SELECT prw.* FROM pa_notification_service.product_mails prw WHERE 1=1  
       and prw.product_id=$1	
       and ((user_id=$2 and to_user_id=$3) OR (user_id=$3 and to_user_id=$2))
  	   and to_user_id<>0
  	   and prw.blocked is null and prw.deleted is null 
  	order by created asc `,
    GET_PRODUCT_MAIL_PERSONAL:  `
    SELECT * FROM pa_notification_service.product_mails_media_storage WHERE mail_id=$1 and blocked is null and deleted is null`,
    SET_PRODUCT_MAIL:  `
    INSERT INTO product_mails (product_id, user_id, comment, to_user_id) values ($1,$2,$3,$4) ON DUPLICATE KEY UPDATE comment=$5 RETURNING id`,
    SET_MAIL_IMAGE:  `
    INSERT INTO product_mails_media_storage (mail_id, media_id, media_key, product_id, user_id, storage, bucket) VALUES ($1,$2,$3,$4,$5,$6,$7)  RETURNING id`,
  },  
};

const MESSAGES = {
   RU : {    
      INPUT_VALIDATION_ERROR: 'Ошибка проверки входных данных',
   }
}

/* Экспорт */
module.exports = {
  SQL, MESSAGES
};
